docs/faucets/dd.md

```diff
@@ -0,0 +1,418 @@
+# Testnet Faucet Design Document
+
+- Owner: @just-mitch
+- Approvers:
+  - @aminsammara
+  - @rahul-kothari
+  - @LHerskind
+  - @PhilWindle
+  - @signorecello
+- [PRD](https://github.com/AztecProtocol/engineering-designs/blob/11ff0c45015027554f13d37f2df85e84090a55a4/docs/faucets/prd.md)
+- Target DD Approval Date: 2025-03-20
+- Target Project Delivery Date: 2025-03-28
+
+## Executive Summary
+
+Update the fee asset and staking asset to have "minter" roles.
+
+Deploy new "handler" contracts that have the minter role on their respective assets.
+
+The FeeAssetHandler will mint a fixed amount to any caller.
+
+The StakingAssetHandler will add an address to the validator set, when called by the owner.
+
+The discord bot should be augmented to allow someone to pass it a merkle proof of the archive tree, and only then be added to the validator set.
+
+## Timeline
+
+Outline the timeline for the project. E.g.
+
+- Core contract dev : 1-2 days
+  - build fee asset handler
+  - build staking asset handler
+  - tests
+- discord bot dev : 1-2 days
+- run upgrade : <0.5 days
+
+Total: 3-5 days
+
+## Introduction
+
+We are not concerned about the DOS risk of someone holding a large amount of fee asset and flooding the network with transactions: the node and p2p layer should be able to handle this; controlling the production of fee asset needs only to prevent infinite minting (and thus a DOS against the users of the network).
```

Comment:
Is this true though? That we aren't concerned with users minting lots of fee asset and then flooding the network? 

Assuming the node/p2p can cope with it. Doesn't it just do 1 of 2 things?

1. Build up a huge backlog of pointless txs from bots/farmers that hogs the network for a long time.
2. If we set the mempool limits low to avoid a build up, people have their txs evicted. Either way the user exp is terriblke.